### DISCUSSION TITLE
ScriptReference/SceneManagement.SceneUtility.GetScenePathByBuildIndex

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#660] [Notes]
# ScriptReference/SceneManagement.SceneUtility.GetScenePathByBuildIndex

https://docs.unity3d.com/ScriptReference/SceneManagement.SceneUtility.GetScenePathByBuildIndex.html

**Comment by andreasIsCoding** [comment]
Note that this method will return an incorrect path if used in the editor, and you have disabled scenes in the build settings above the scene you want to get.  Here is a method that correctly gets the scene name by build index (editor only, tested in 2020.3.48f):
```
public static string getRealPathByBuildI (int sceneI)
{
	int i = 0;
	foreach (EditorBuildSettingsScene sceneInfo in EditorBuildSettings.scenes)
	{
		if (!sceneInfo.enabled) continue;
		
		if (i == sceneI) return sceneInfo.path;

		i++;
	}

	return null;
}
```
